src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessModuleTests.cs

```diff
@@ -18,26 +19,30 @@ public void TestModulePropertiesExceptOnOSX()
             if (global::Interop.IsWindows)
                 fileName = string.Format("{0}.exe", CoreRunName);
 
-            // Ensure the process has loaded the modules.
-            Assert.True(SpinWait.SpinUntil(() =>
+            try
             {
-                if (_process.Modules.Count > 0)
-                    return true;
-                _process.Refresh();
-                return false;
-            }, WaitInMS));
+                // Ensure the process has loaded the modules.
+                Assert.True(SpinWait.SpinUntil(() =>
+                {
+                    if (_process.Modules.Count > 0)
+                        return true;
+                    _process.Refresh();
+                    return false;
+                }, WaitInMS));
 
-            ProcessModuleCollection moduleCollection = _process.Modules;
-            Assert.True(moduleCollection.Count > 0);
+                ProcessModuleCollection moduleCollection = _process.Modules;
+                Assert.True(moduleCollection.Count > 0);
 
-            ProcessModule coreRunModule = _process.MainModule;
-            Assert.True(coreRunModule.BaseAddress.ToInt64() > 0);
-            Assert.True(coreRunModule.EntryPointAddress.ToInt64() >= 0);
-            Assert.Equal(fileName, coreRunModule.ModuleName);
-            Assert.True(coreRunModule.ModuleMemorySize > 0);
-            Assert.EndsWith(fileName, coreRunModule.FileName);
+                ProcessModule coreRunModule = _process.MainModule;
+                Assert.True(coreRunModule.BaseAddress.ToInt64() > 0);
+                Assert.True(coreRunModule.EntryPointAddress.ToInt64() >= 0);
+                Assert.Equal(fileName, coreRunModule.ModuleName);
+                Assert.True(coreRunModule.ModuleMemorySize > 0);
+                Assert.EndsWith(fileName, coreRunModule.FileName);
 
-            Assert.Equal(string.Format("System.Diagnostics.ProcessModule ({0})", fileName), coreRunModule.ToString());
+                Assert.Equal(string.Format("System.Diagnostics.ProcessModule ({0})", fileName), coreRunModule.ToString());
+            }
+            catch (Win32Exception) { } // expected, known behavior of Process.MainModule to throw, when 32 bit process accesses a 64bit PEB on 64 bit machines.
```

Comment:
As per the documentation, if the test process is 64 bit then this error should not happen right as @stephentoub suggests.. A 64 bit process can access 64bit TestConsoleApp.exe process modules, as well as, 32 bit TestConsoleApp.exe process modules.. Or better yet, i can test on the CurrentProcess Modules, instead of testing on another process, then this problem wouldn't occur.